pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -720,9 +720,15 @@ self: super: {
   });
 
   # The standard libraries are compiled separately
-  idris = generateOptparseApplicativeCompletion "idris" (
-    doJailbreak (dontCheck super.idris)
-  );
+  idris = generateOptparseApplicativeCompletion "idris" (doJailbreak (dontCheck
+    (appendPatches
+      (super.idris.override { megaparsec = self.megaparsec_7_0_5; }) [
+        (pkgs.fetchpatch {
+          url = "https://github.com/idris-lang/Idris-dev/pull/4808.diff";
+          sha256 = "060ib1rczy34ip8xf3bv1pf28655f6s0bvvij19jhh5dpcr0pf71";
+          excludes = [ ".travis.yml" "Makefile" "appveyor.yml" ];
+        })
```

Comment:
Could you add a comment to both the `override` and the `fetchpatch` lines describing why they are needed, linking to any relevant upstream issues/PRs, and adding a guess about when they can be removed (when a new upstream version is released??).

Also, please rebase this onto `haskell-updates` and change the base branch on Github to `haskell-updates`.

Reply:
Done.  Thanks!